sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py

```diff
@@ -301,6 +310,11 @@ def get_directory_properties(self, **kwargs):
                 :dedent: 4
                 :caption: Getting the properties for a file/directory.
         """
+        upn = kwargs.pop('upn', None)
+        if upn:
+            headers = kwargs.pop('headers', {})
+            headers['x-ms-upn'] = str(upn)
+            kwargs['headers'] = headers
         return self._get_path_properties(cls=deserialize_dir_properties, **kwargs)  # pylint: disable=protected-access
```

Comment:
So now that directory is included, I think you can move the `upn` logic into `_get_path_properties` of `PathClient` (and async). That way its shared for file/directory.